### ISSUE TITLE
`--single-step` option not working with packed workflow

### ISSUE THREAD

**Issue by jdidion** [common-workflow-language/cwltool#1443]
`cwltool --debug --single-step ext test/cwl_conformance/workflows_packed/basename-fields-test.cwl.json input.json`

with input 

```json
{
  "in": ".cwl",
  "name": null
}
```

gives the following error:

```
INFO /Users/jdidion/.pyenv/versions/dxwdl/bin/cwltool 3.0.20210319143721
INFO Resolved 'test/cwl_conformance/workflows_packed/basename-fields-test.cwl.json' to 'file:///Users/jdidion/projects/dxCompiler/test/cwl_conformance/workflows_packed/basename-fields-test.cwl.json'
ERROR I'm sorry, I couldn't load this CWL file.
The error was: 
Traceback (most recent call last):
  File "/Users/jdidion/.pyenv/versions/3.6.9/envs/dxwdl/lib/python3.6/site-packages/cwltool/main.py", line 1073, in main
    ctool = choose_step(args, tool, loadingContext)
  File "/Users/jdidion/.pyenv/versions/3.6.9/envs/dxwdl/lib/python3.6/site-packages/cwltool/main.py", line 815, in choose_step
    extracted = get_step(tool, tool.tool["id"] + "/" + args.singe_step)
AttributeError: 'Namespace' object has no attribute 'singe_step'
```

The workflow is attached.

[basename-fields-test.cwl.txt](https://github.com/common-workflow-language/cwltool/files/6381869/basename-fields-test.cwl.txt)

**Comment by mr-c** [comment]
Hey @jdidion ; I've fixed this typo in #1444

As for the input object, looks like you need to smoosh the workflow level and the step level inputs together. Perhaps this was done to support scattering?

So the following `input.json` works with the PR for me
```
{
  "tool": {"class": "File", "path": "example.cwl"},
  "in": ".cwl"
}
```

**Comment by jdidion** [comment]
It looks like it ignores the value of `in` and evaluates the `valueFrom` expression to get the value of `in`. It should be either/or - either `in` is specified in the input and that value is used, or `tool` is specified and the `valueFrom` is evaluated to get the value.
